pkg/storage/tsdb/bucketcache/caching_bucket.go

```diff
@@ -159,6 +159,38 @@ func NewCachingBucket(bucketID string, bucketClient objstore.Bucket, cfg *Cachin
 	return cb, nil
 }
 
+func (cb *CachingBucket) invalidate(ctx context.Context, name string) {
```

Comment:
[nit] I would add a comment to explain this is just a best effort. It suffers race conditions, and there's no guarantee that an upload/delete that was reported as failure wasn't actually successfully completed on the server side.